source/numerics/data_out_resample.cc

```diff
@@ -0,0 +1,178 @@
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_tools.h>
+
+#include <deal.II/numerics/data_out_dof_data.templates.h>
+#include <deal.II/numerics/data_out_resample.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <sstream>
+
+DEAL_II_NAMESPACE_OPEN
+
+
+template <int dim, int patch_dim, int spacedim>
+DataOutResample<dim, patch_dim, spacedim>::DataOutResample(
+  const Triangulation<patch_dim, spacedim> &patch_tria,
+  const Mapping<patch_dim, spacedim> &      patch_mapping)
+  : patch_tria(patch_tria)
+  , patch_dof_handler(patch_tria)
+  , patch_mapping(patch_mapping)
+{}
+
+
+
+template <int dim, int patch_dim, int spacedim>
+void
+DataOutResample<dim, patch_dim, spacedim>::update_mapping(
+  const Mapping<dim, spacedim> &mapping,
+  const unsigned int            n_subdivisions)
+{
+  this->mapping = &mapping;
+  this->indices.clear();
+
+  FE_Q<patch_dim, spacedim> fe(std::max<unsigned int>(1, n_subdivisions));
+  patch_dof_handler.distribute_dofs(fe);
+
+  std::vector<std::pair<types::global_dof_index, Point<spacedim>>> points_all;
+
+  QGaussLobatto<patch_dim> quadrature_gl(fe.degree + 1);
+
+  std::vector<Point<patch_dim>> quadrature_points;
+  for (const auto i :
+       FETools::hierarchic_to_lexicographic_numbering<patch_dim>(fe.degree))
+    quadrature_points.push_back(quadrature_gl.point(i));
+  Quadrature<patch_dim> quadrature(quadrature_points);
+
+  FEValues<patch_dim, spacedim> fe_values(patch_mapping,
+                                          fe,
+                                          quadrature,
+                                          update_quadrature_points);
+
+  std::vector<types::global_dof_index> dof_indices(fe.n_dofs_per_cell());
+
+  IndexSet active_dofs;
+  DoFTools::extract_locally_active_dofs(patch_dof_handler, active_dofs);
+  partitioner = std::make_shared<Utilities::MPI::Partitioner>(
+    patch_dof_handler.locally_owned_dofs(), active_dofs, MPI_COMM_WORLD);
+
+  for (const auto &cell : patch_dof_handler.active_cell_iterators())
+    {
+      if (cell->is_locally_owned() == false)
+        continue;
+
+      fe_values.reinit(cell);
+
+      cell->get_dof_indices(dof_indices);
+
+      for (unsigned int i = 0; i < dof_indices.size(); ++i)
+        points_all.emplace_back(partitioner->global_to_local(dof_indices[i]),
+                                fe_values.quadrature_point(i));
+    }
+
+  std::sort(points_all.begin(),
+            points_all.end(),
+            [](const auto &a, const auto &b) { return a.first < b.first; });
```

Comment:
don't you also need to consider the case that `a.first == b.first`?

Reply:
Why? std::sort expects a lambda for `<`. In the next step, when I remove duplicates I am checking `==`.

Reply:
I was just imagining that you are doing a lexicographic ordering. But maybe you don't :-)